src/parser/warrior/arms/modules/core/Overpower.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+
+import { formatPercentage } from 'common/format';
+import SpellLink from 'common/SpellLink';
+import SPELLS from 'common/SPELLS';
+import Analyzer from 'parser/core/Analyzer';
+
+import Enemies from 'parser/shared/modules/Enemies';
+import ExecuteRange from './Execute/ExecuteRange';
+import SpellUsable from '../features/SpellUsable';
+
+class OverpowerAnalyzer extends Analyzer {
+    static dependencies = {
+        executeRange: ExecuteRange,
+        enemies: Enemies,
+        spellUsable: SpellUsable,
+    }
+
+    proc = 0;
+    wastedProc = 0;
+
+    on_byPlayer_cast(event) {
+        if (event.ability.guid !== SPELLS.OVERPOWER.id) {
+            return;
+        }
+
+        this.proc += 1;
+        const overpower = this.selectedCombatant.getBuff(SPELLS.OVERPOWER.id);
+        if (!this.executeRange.isTargetInExecuteRange(event) && !this.selectedCombatant.hasTrait(SPELLS.EXECUTIONERS_PRECISION_TRAIT.id)) {
```

Comment:
Not incorrect, although if you want to, you can reduce the verbosity a little by setting a constant in constructor, like 
```javascript
this.hasEP = this.selectedCombatant.hasTrait(SPELLS.EXECUTIONERS_PRECISION.id)
```